polkadot/runtime/rococo/src/lib.rs

```diff
@@ -1826,6 +1826,12 @@ sp_api::impl_runtime_apis! {
 
 	impl offchain_primitives::OffchainWorkerApi<Block> for Runtime {
 		fn offchain_worker(header: &<Block as BlockT>::Header) {
+			use sp_runtime::{traits::Header, DigestItem};
+
+			if header.digest().logs().iter().any(|di| di == &DigestItem::RuntimeEnvironmentUpdated) {
```

Comment:
this will not prevent running the migration twice if there are two upgrades happening, just reduce it to runtime upgrades, no?
I guess it doesn't really matter since cleaning should be idempotent

Reply:
Yeah, as far as I see from the docs, those `RuntimeEnvironmentUpdated` could be triggered even not only by runtime upgrades but by some other environment changes as well, but we don't care. In the worst case, it will try to clear already cleared keys once more, nothing will explode.